test/exercises/2023JanTutorial/kmer.chpl

```diff
@@ -0,0 +1,48 @@
+// kmer.chpl
+/*
+   usage on puma/ocelote:
+     chpl kmer.chpl
+     ./kmer
+
+   usage on laptop with podman (or docker):
+     podman run --rm -v "$PWD":/myapp -w /myapp chapel/chapel chpl kmer.chpl
+     podman run --rm -v "$PWD":/myapp -w /myapp chapel/chapel ./kmer
+
+   For docker usage, see https://chapel-lang.org/install-docker.html
+
+   Original version of kmer counting algorithm provided by
+   Aryamaan Jain (github strikeraryu) on Chapel Discourse April 7, 2021.
+
+   kmer_large_input.txt is from
+   https://www.ncbi.nlm.nih.gov/nuccore/NC_001422.1?report=fasta
+ */
+
+use Map;
+use IO;
+
+// to have it read a different input file, run as follows:
+//      ./kmer --infile="anotherFileName"
+config const infile = "kmer_large_input.txt";
+
+// set k to something different on the commandline with
+//      ./kmer --k=7
+config const k = 4;
+
+// read in the input sequence from the file infile
+var sequence : string;
+var f = open(infile, iomode.r);
+var fReader =  f.reader();
+fReader.read(sequence);
+
+// declare a dictionary/map to store the count per kmer
+var nkmerCounts : map(string, int);
+
+// count up the number of times each kmer occurs
+for ind in 0..<(sequence.size-k) {
+  nkmerCounts[sequence[ind..<ind+k]] += 1;
```

Comment:
I'd write this slice as `ind..#k` which is a perfect rationale for why `#` is in the language.  If you didn't want to teach both `..<` and `#` in this one benchmark, this is a case where I'd actually switch the former loop to `#` rather than this slice as `..<`.

Reply:
I agree with you that ind..#k makes sense, but yesterday I thought you indicated the other one would be better with <?  Or maybe that was in another code?

Reply:
Yeah, I'm pretty sure that was the diffusion code, which was written `0..#n` where I was suggesting `0..<n`.  In general, I prefer `lo..<hi` ("from lo to hi, excluding it") when trying to express `lo..hi-1`, particularly for 0-based indexing.  For a `lo..lo+k-1` pattern, I prefer `lo..#k` ("start at lo and give me k things").